### PR TITLE
Fix #12: Adds faker gem

### LINKED ISSUE #12: Add Faker gem
Add Faker gem

Reference: https://youtu.be/nDpnb8Nm0Wk

### PR DESCRIPTION
**Pull request by vanamraghu**
Fixes #12 with the below changes

1. Modified ruby version to 3.3.4
2. Gemfile updated with 3.3.4 version and also faker gem is added
3. Gemfile.lock is updated with all the versions

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-3.3.2
+3.3.4
```
Comment (rt4914) [inline .ruby-version:1]: Don't update this here. 
Changes like these should have a separate PR on its own as it affects everyone. 
This is the ideal process:
1. File an issue this task.
2. Create a Pull Request
3. Tag everyone (all developers) on that pull request so that everyone can confirm it works for them and they are okay with it. 
4. Once everyone approves, then we can merge it.


These steps are important so as to make sure everyone is comfortable and not a single developer is left-out.

File: Gemfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 source "https://rubygems.org"
 
-ruby "3.3.2"
+ruby "3.3.4"
```
Comment (rt4914) [inline Gemfile:3]: Undo this change too.

File: Gemfile [hunk 3]
```diff
@@ -47,6 +47,8 @@ gem "bootsnap", require: false
 group :development, :test do
   # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
   gem "debug", platforms: %i[ mri windows ]
+
+  gem 'faker', :git => 'https://github.com/faker-ruby/faker.git', :branch => 'main'
```
Comment (rt4914) [inline Gemfile:51]: Simply use
`gem 'faker'` and it should be fine

File: Gemfile.lock [hunk 4]
```diff
@@ -1,3 +1,11 @@
+GIT
```
Comment (rt4914) [inline Gemfile.lock:1]: Regenerate this file again as per above changes.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@vanamraghu Please update the title to

`Fix #12: Adds faker gem`

File: .ruby-version (hunk 1 above)
Comment (vanamraghu) [reply .ruby-version:1]: Reverting the change to 3.3.2

File: Gemfile.lock (hunk 4 above)
Comment (vanamraghu) [reply Gemfile.lock:1]: Done

**Review by rt4914 — APPROVED** [review]
LGTM. Thanks.

One thing to keep in mind, the person who starts the comment-thread will resolve that thread. So basically in my last review I left few comments, you replied to those comments which is correct but you also resolved them, instead of that I am suppose to verify as per your replies and if I am satisfied I will resolve them.
Hope this clears things.
